crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -334,92 +332,30 @@ where
         Ok(token.as_ref().to_string())
     }
 
-    // Find the given IBC message in the changed keys and return the associated
-    // sequence number
-    fn find_ibc_transfer_sequence(
-        &self,
-        message: &IbcMsgTransfer,
-        keys_changed: &BTreeSet<Key>,
-    ) -> Result<Option<Sequence>> {
-        // Compute the packet commitment for this message
-        let packet_data_bytes = serde_json::to_vec(&message.packet_data)
-            .map_err(native_vp::Error::new)?;
-        let packet_commitment =
-            VpValidationContext::<'a, 'a, S, CA>::compute_packet_commitment(
-                &packet_data_bytes,
-                &message.timeout_height_on_b,
-                &message.timeout_timestamp_on_b,
-            );
-        // Try to find a key change with the same port, channel, and commitment
-        // as this message and note its sequence number
-        for key in keys_changed {
-            let Some(path) = is_ibc_commitment_key(key) else {
-                continue;
-            };
-            if path.port_id == message.port_id_on_a
-                && path.channel_id == message.chan_id_on_a
-            {
-                let Some(storage_commitment): Option<PacketCommitment> =
-                    self.ctx.read_bytes_post(key)?.map(Into::into)
-                else {
-                    // Ignore this event if it does not exist
-                    continue;
-                };
-                if packet_commitment == storage_commitment {
-                    return Ok(Some(path.sequence));
-                }
-            }
-        }
-        Ok(None)
-    }
-
-    // Apply the given transfer message to the changed balances structure
+    // Apply the given send packet to the changed balances structure
     fn apply_transfer_msg(
         &self,
         mut acc: ChangedBalances,
-        msg: &IbcMsgTransfer,
-        keys_changed: &BTreeSet<Key>,
+        src_port_id: &PortId,
+        src_channel_id: &ChannelId,
+        ibc_trace: impl AsRef<str>,
+        amount: Amount,
+        receiver: impl AsRef<str>,
     ) -> Result<ChangedBalances> {
-        // If a key change with the same port, channel, and commitment as this
-        // message cannot be found, then ignore this message. Though this check
-        // is done in the IBC VP, the test is repeated here to avoid making
-        // assumptions about how the IBC VP interprets the given message.
-        if self
```

Comment:
Could we keep this check or have something with similar effect so that we avoid making assumptions about the execution of the IBC VP? See also https://github.com/anoma/namada/pull/3409#discussion_r1644620706 .